data-prepper-api/src/main/java/org/opensearch/dataprepper/model/acknowledgements/AcknowledgementSet.java

```diff
@@ -58,4 +61,16 @@ public interface AcknowledgementSet {
       * initial events are going through the pipeline line.
      */
     public void complete();
+
+    /**
+     * adds progress check callback to the acknowledgement set. When added
+     * the callback is called every progressCheckInterval time with the 
+     * indication of current progress as a ratio of pending number of
+     * acknowledgements over total acknowledgements
+     *
+     * @param progressCheckCallback progress check callback to be called
+     * @param progressCheckInterval frequency of invocation of progress check callback
+     * @since 2.6
+     */
+    public void addProgressCheck(final Consumer<Double> progressCheckCallback, final Duration progressCheckInterval);
```

Comment:
I'm curious why you chose to make this a `Consumer<Double>`. The current code isn't using this ratio presently.

A few alternatives:

* Make this a `Runnable` without any input.
* Create a new interface that can be provided so that the relevant arguments can be expanded in the future. For example, `Consumer<ProgressCheck>`.

```
public interface ProgressCheck {
  Double getRatio();
}
```

Then we could add other items if desired such as `Duration getTimeToExpiry()`.